examples/autoscalinggroup/main.tf

```diff
@@ -0,0 +1,122 @@
+################################
+# terraform-null-label example #
+################################
+module "label" {
+  source    = "../../"
+  namespace = "awesomeproject"
+  stage     = "production"
+  name      = "clusterpluck"
+
+  tags = {
+    BusinessUnit = "Finance"
+    ManagedBy    = "Terraform"
+  }
+
+  additional_tag_map = {
+    propagate_at_launch = "true"
+  }
+}
+
+#######################
+# Launch template     #
+#######################
+resource "aws_launch_template" "this" {
+  # terraform-null-label example used here: Set template name prefix
+  name_prefix                          = "${module.label.id}-"
+  image_id                             = "${data.aws_ami.amazon_linux.id}"
+  instance_type                        = "t2.micro"
+  instance_initiated_shutdown_behavior = "terminate"
+
+  vpc_security_group_ids = ["${data.aws_security_group.default.id}"]
+
+  monitoring {
+    enabled = false
+  }
+
+  # terraform-null-label example used here: Set tags on volumes
+  tag_specifications {
+    resource_type = "volume"
+    tags          = "${module.label.tags}"
+  }
+}
+
+######################
+# Autoscaling group  #
+######################
+resource "aws_autoscaling_group" "this" {
```

Comment:
We typically use `default` in place of `this`. Using `this` is also a good convention, but let's stick with `default` since that's what we have else where.